src/components/FileTreeNode/index.spec.tsx

```diff
@@ -216,4 +246,200 @@ describe(__filename, () => {
+        },
+      });
+
+      expect(root.find(className)).toHaveLength(1);
+
+      const nodeIcons = root.find(`.${styles.nodeIcons}`);
+      expect(nodeIcons).toHaveLength(1);
+      expect(nodeIcons.find(FontAwesomeIcon)).toHaveProp('icon', icon);
+      expect(nodeIcons.find(FontAwesomeIcon).prop('title')).toMatch(
+        new RegExp(`contains files with linter ${type}s`),
+      );
+    },
+  );
+
+  it('ignores linter messages unrelated to the current node', () => {
+    const message = {
+      ...fakeExternalLinterMessage,
+      file: 'manifest.json',
+    };
+
+    const root = renderWithLinterMessage({
+      message,
+      treefoldRenderProps: {
+        id: 'some/other/file.js',
+      },
+    });
+
+    expect(root.find(`.${styles.hasLinterMessages}`)).toHaveLength(0);
+  });
+
+  it('indicates when a directory node contains a file that has linter messages', () => {
+    const message = {
+      ...fakeExternalLinterMessage,
+      file: 'src/manifest.json',
+    };
+
+    const root = renderWithLinterMessage({
+      message,
+      treefoldRenderProps: {
+        id: 'src/',
+        isFolder: true,
+      },
+    });
+
+    expect(root.find(`.${styles.hasLinterMessages}`)).toHaveLength(1);
+  });
+
+  describe('findMostSevereTypeForPath', () => {
+    it('returns null when the map is empty', () => {
+      const path = 'file.js';
+      const map = _getMessageMap([]);
+
+      const type = findMostSevereTypeForPath(map, path);
+
+      expect(type).toEqual(null);
+    });
+
+    it('returns the type of the message when there is only one message and a path', () => {
+      const path = 'file.js';
+      const message = { ...fakeExternalLinterMessage, file: path };
+      const map = _getMessageMap([message]);
+
+      const type = findMostSevereTypeForPath(map, path);
+
+      expect(type).toEqual(message.type);
+    });
+
+    it('returns null when the path is not found', () => {
+      const path = 'file.js';
+      const message = { ...fakeExternalLinterMessage, file: path };
+      const map = _getMessageMap([message]);
+
+      const type = findMostSevereTypeForPath(map, 'other-path');
+
+      expect(type).toEqual(null);
+    });
+
+    it('returns the most severe type given a directory path', () => {
+      const path = 'src';
+      const message = { ...fakeExternalLinterMessage, file: `${path}/file.js` };
+      const map = _getMessageMap([message]);
+
+      const type = findMostSevereTypeForPath(map, path);
+
+      expect(type).toEqual(message.type);
+    });
+
+    it('returns the most severe type given a map of messages and a directory path', () => {
+      const path = 'src';
+      const messages = [
+        {
+          ...fakeExternalLinterMessage,
+          file: `${path}/file-1.js`,
+          type: 'warning',
+        },
+        {
+          ...fakeExternalLinterMessage,
+          file: `${path}/file-2.js`,
+          type: 'error',
+        },
+      ] as ExternalLinterMessage[];
```

Comment:
This is not a change request but since this has come up a lot, maybe we should switch to string enums for `type`.

Reply:
it's a bummer that it does not understand this notation...